segmentation_models_pytorch/encoders/timm_gernet.py

```diff
@@ -0,0 +1,120 @@
+from timm.models import ByobCfg, BlocksCfg, ByobNet
+
+from ._base import EncoderMixin
+import torch.nn as nn
+
+
+class GERNetEncoder(ByobNet, EncoderMixin):
+    def __init__(self, out_channels, depth=6, **kwargs):
```

Comment:
For genet I've found 6(+rgb=7) stages and most others encoders have 5, maybe I've misunderstood the concept.

Reply:
each stage should reduce spatial resolution, you may combine several stages in one

Reply:
How should we combine stages if the constant resolution stages are different for each version?
(3, 13, 48, 48, 384, 560, 1920) - s 
(3, 32, 128, 192, 640, 640, 2560) - m, l

Reply:
this is channels resolution, not spatial
try to print tensor shapes afrer each stage

Reply:
suppose you have image 224 224 3
stem 112 112 13
stage_1 64 64 48
stage_2 32 32 48
stage_3 32 32 386
etc...

so, you should combine stage_2 and stage_3

Reply:
```
s - [torch.Size([1, 3, 64, 64]), torch.Size([1, 13, 32, 32]), torch.Size([1, 48, 16, 16]), torch.Size([1, 48, 8, 8]), torch.Size([1, 384, 4, 4]), torch.Size([1, 560, 2, 2]), torch.Size([1, 1920, 2, 2])]
m, l - [torch.Size([1, 3, 64, 64]), torch.Size([1, 32, 32, 32]), torch.Size([1, 128, 16, 16]), torch.Size([1, 192, 8, 8]), torch.Size([1, 640, 4, 4]), torch.Size([1, 640, 2, 2]), torch.Size([1, 2560, 2, 2])]
```
Last 2 should be joined as I understood?

Reply:
Yes, and `out_channels` should be modified:
 (3, 13, 48, 48, 384, 560, 1920)  ->
 (3, 13, 48, 48, 384, 1920)

Reply:
And depth to 5